tmt/steps/report/polarion.py

```diff
@@ -345,6 +345,14 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
             }
         )
 
+        # Add deployment mode if provided as a context variable
+        try:
+            deployment_mode = self.step.plan._fmf_context['deployment-mode'][0]
+        except Exception:
+            deployment_mode = None
+        if deployment_mode:
+            testsuites_properties.update({'polarion-deploymentmode': deployment_mode})
```

Comment:
Since I really dislike exceptions, how about the following? It would have the advantage of not silencing all possible errors:

```python
deployment_mode = self.step.plan._fmf_context.get('deployment-mode', [])

if deployment_mode:
    testsuites_properties.update({'polarion-deploymentmode': deployment_mode[0]})
```